### ISSUE TITLE
Add update_metadata convenience function

### ISSUE THREAD

**Issue by constantinpape** [constantinpape/pybdv#14]
cc @martinschorb

**Comment by martinschorb** [comment]
Hi,

when I run make_bdv with overwrite=False and the specified view setup already exists, it should do exactly that: update the metadata only. Is that what it anyway does? Otherwise it should in my opinion...

**Comment by constantinpape** [comment]
> when I run make_bdv with overwrite=False and the specified view setup already exists, it should do exactly that: update the metadata only. Is that what it anyway does?

No, currently it does not do this, but just skips the corresponding setup if it exists already.

> Otherwise it should in my opinion...

I am not so sure. As a user, I would expect that neither data nor metadata changes if `overwrite=False`.

Let's discuss this in person at some point after Easter.

**Comment by constantinpape** [comment]
Ok, I thought about this a bit more, and I tend to agree with you that this should just be handled by `make_bdv` (or `convert_to_bdv`).
However, I think we should have multiple modes, something like:
- `overwrite-data`
- `overwrite-metadata`
- `skip` (= over-write neither data nor metadata)
- `overwrite-data-and-metadata`

The default would be `skip` to be backward compatible.
These could be passed as strings or integers.
Let's discuss details tomorrow.

**Comment by martinschorb** [comment]
[fm_r1.zip](https://github.com/constantinpape/pybdv/files/4491984/fm_r1.zip)

**Comment by constantinpape** [comment]
Ok, I have implemented more fine-grained over-write modes `skip`, `data`, `metadata` and `all` now.
@martinschorb please give it a try and let me know if that works for you.

**Comment by martinschorb** [comment]
Hi,

I run with ``` overwrite = 'metadata'```
setupid exists, so we should be clearly in case4 :
```
    # 4: setup id and time-point are present and over-write mode is 'metadata' -> don't over-write data,
    #                                                                             over-write metadata
```

it still writes new h5 files...

**Comment by constantinpape** [comment]
Ok, I will update my tests to check for this case as well and see if I can catch it and then also fix it.

**Comment by constantinpape** [comment]
I can reproduce the issue and have added a unit-test.
Will fix it later.

**Comment by constantinpape** [comment]
> Hi,
> 
> I run with ` overwrite = 'metadata'`
> setupid exists, so we should be clearly in case4 :
> 
> ```
>     # 4: setup id and time-point are present and over-write mode is 'metadata' -> don't over-write data,
>     #                                                                             over-write metadata
> ```
> 
> it still writes new h5 files...

This should work now.
@martinschorb can you check again?

**Comment by martinschorb** [comment]
Hmm,

now it even does more strange things...

```metadata```:

- writes h5 files all the time calling ```make_bdv```
- adds additional ```ViewRegistrations``` for existing ```ViewSetups``` instead of overwriting those, even though it tells me: ```Setup id 0 is present already and will be over-written```
but produces:

```
 <ViewRegistrations>
    <ViewRegistration setup="0" timepoint="0">
      <ViewTransform type="affine">
        <affine>1.0 0.0 0.0 43.1844 0.0 1.0 0.0 131.3537 0.0 0.0 1.0 13.061</affine>
        <Name>Amira_Translation</Name>
      </ViewTransform>
      <ViewTransform type="affine">
        <affine>0.666346 0.0 0.0 0.0 0.0 0.666079 0.0 0.0 0.0 0.0 0.326526 0.0</affine>
        <Name>Amira_Transform</Name>
      </ViewTransform>
    </ViewRegistration>
    <ViewRegistration setup="0" timepoint="0">
      <ViewTransform type="affine">
        <affine>1.0 0.0 0.0 43.1844 0.0 1.0 0.0 131.3537 0.0 0.0 1.0 13.061</affine>
        <Name>Amira_Translation</Name>
      </ViewTransform>
      <ViewTransform type="affine">
        <affine>0.666346 0.0 0.0 0.0 0.0 0.666079 0.0 0.0 0.0 0.0 0.326526 0.0</affine>
        <Name>Amira_Transform</Name>
      </ViewTransform>
    </ViewRegistration>
  </ViewRegistrations>
```
